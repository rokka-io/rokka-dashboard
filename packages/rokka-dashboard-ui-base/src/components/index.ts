export * from './formGroup/formGroup';
export * from './loadingIndicatingButton/loadingIndicatingButton';
export * from './signForm/signForm';
export * from './user/user';
