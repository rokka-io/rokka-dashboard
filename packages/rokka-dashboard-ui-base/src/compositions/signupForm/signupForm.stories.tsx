import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React,  { useState } from 'react';
import { colors } from '../../identity/colors/colors';
import { SignupForm, SuccessCb } from './signupForm';

const SignupContainer = () => {
  const [loggedin, setSignedin] = useState(false);
  const onSignup: (organization: string, email: string, successCb: SuccessCb) => Promise<void> = (
    _organization: string,
    _email: string,
    successCb: SuccessCb
  ) => {
    const done = () => {
      setSignedin(true);
    };
    return new Promise(resolve => {
      setTimeout(() => {
        successCb(done);
        resolve();
      }, 500);
    });
  };

  if (loggedin) {
    setTimeout(() => {
      setSignedin(false);
    }, 2000);
    return <span>Signed up</span>;
  }
  return <SignupForm onSignup={onSignup} />;
};

storiesOf('Compositions', module)
  // TODO: decide if we want to move that to a separate decorator so it can be used somewhere else, too.
  .addDecorator(story => (
    <div style={{ position: 'relative', width: '100vw', minHeight: '100vh', background: colors.gray.lightest }}>
      {story()}
    </div>
  ))
  .add('SignupForm', () => <SignupContainer />);
