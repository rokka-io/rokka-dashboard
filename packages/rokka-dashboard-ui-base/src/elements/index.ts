export * from './button/button';
export * from './headerLogo/headerLogo';
export * from './heading1/heading1';
export * from './heading2/heading2';
export * from './icon/icon';
export * from './input/input';
export * from './label/label';
export * from './link/link';
export * from './paragraph/paragraph';
export * from './spinner/spinner';
