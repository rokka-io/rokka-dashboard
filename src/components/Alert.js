import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'

const Alert = ({ alert = null }) => {
  if (!alert) {
    return null
  }

  let message = alert.message
  if (Array.isArray(message)) {
    message = message.map((msg, index) => (<span className="rka-alert-msg" key={index}>{msg}</span>))
  } else {
    message = (<span className="rka-alert-msg">{message}</span>)
  }

  return <div className={cx('rka-alert', `is-${alert.type}`, 'txt-c')}>{message}</div>
}
Alert.propTypes = {
  alert: PropTypes.shape({
    type: PropTypes.string,
    message: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)])
  })
}

export default Alert
