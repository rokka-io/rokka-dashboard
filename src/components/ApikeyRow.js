import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import rokka from '../rokka'
import { formatDate } from '../utils/string'

const DEFAULT_STATE = {
  currentKeyId: '',
  delete: false
}

class ApikeyRow extends PureComponent {
  constructor(props) {
    super(props)
    this.state = DEFAULT_STATE
  }
  showDelete = () => {
    this.setState({ delete: true })
  }

  reallyDelete = () => {
    rokka()
      .user.deleteApiKey(this.props.apiKey.id)
      .then(({ body }) => {
        this.props.updateKeys()
      })
      .catch(err => {
        alert("Api Key deletion didn't work:" + err.body.error.message)
      })
  }
  render() {
    return (
      <tr key={this.props.apiKey.id}>
        <td className={'mb-md'} style={{ height: '4em' }}>
          {this.props.apiKey.id}
        </td>
        <td>{this.props.apiKey.comment}</td>
        <td>{formatDate(this.props.apiKey.created, 'Before December 2021')}</td>
        <td>{formatDate(this.props.apiKey.accessed, 'Before December 2021')}</td>
        <td>
          {this.props.apiKey.id === this.props.currentKeyId ? (
            <>
              <div className={'mb-sm'}>Currently used Api Key, not deletable </div>
              <div>(use another Api Key to login, if you want to delete this one)</div>
            </>
          ) : this.state.delete ? (
            <>
              <button
                className="rka-button rka-button-brand  mr-md"
                onClick={e => this.reallyDelete(e)}
              >
                Really Delete!
              </button>
              <button
                className="rka-button rka-button-secondary"
                onClick={e => this.setState({ delete: false })}
              >
                Cancel
              </button>
            </>
          ) : (
            <button className="rka-button rka-button-brand" onClick={e => this.showDelete(e)}>
              Delete Api Key
            </button>
          )}
        </td>
      </tr>
    )
  }
}

ApikeyRow.propTypes = {
  apiKey: PropTypes.object.isRequired,
  currentKeyId: PropTypes.string.isRequired,
  updateKeys: PropTypes.func.isRequired
}

export default ApikeyRow
