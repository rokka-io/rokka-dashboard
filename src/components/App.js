import React, { PureComponent } from 'react'
import { Route } from 'react-router'
import { HashRouter } from 'react-router-dom'
import state, { subscribe } from '../state'
import Dashboard from './Dashboard'
import Images from './Images'
import Bills from './Bills'
import Signup from './Signup'
import Stacks from './Stacks'
import NewStack from './NewStack'
import '../scss/main.scss'

class App extends PureComponent {
  constructor(props) {
    super(props)

    this.state = state

    this.unsubscribe = subscribe(state => {
      this.setState(state)
    })
  }

  componentWillUnmount() {
    this.unsubscribe()
  }

  render() {
    return (
      <HashRouter>
        <div className="App">
          <Route path="/" exact render={props => <Dashboard router={props} {...this.state} />} />
          <Route path="/images" render={props => <Images router={props} {...this.state} />} />
          <Route path="/bills" render={props => <Bills router={props} {...this.state} />} />
          <Route path="/signup" render={props => <Signup router={props} {...this.state} />} />
          <Route path="/stacks" render={props => <Stacks router={props} {...this.state} />} />
          <Route path="/new-stack" render={props => <NewStack router={props} {...this.state} />} />
        </div>
      </HashRouter>
    )
  }
}

export default App
