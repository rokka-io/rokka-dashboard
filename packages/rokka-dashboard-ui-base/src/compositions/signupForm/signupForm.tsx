import React, { FormEvent, Fragment, PureComponent } from 'react';
import styled from 'styled-components';
import { FormGroup, LoadingIndicatingButton, SignForm } from '../../components';
import { Heading2, Link, ParagraphWhite, TextInput } from '../../elements';
import { colors } from '../../identity/colors/colors';
import { fonts } from '../../identity/typography';
import { spaces } from '../../identity/spaces/spaces';

export type SignupSuccessCb = (done: () => void) => void;

interface SignupProps {
  /** Callback fired if form is submitted */
  onSignup(organization: string, email: string, successCb: SignupSuccessCb): Promise<void>;
}

interface SignupState {
  /** Entered organization */
  organization?: string;
  /** Entered Email */
  email?: string;
  /** Loading indicator */
  loading?: boolean;
  /** Start transition */
  showTransition?: boolean;
}

export class SignupForm extends PureComponent<SignupProps, SignupState> {
  public state = { organization: '', email: '', loading: false, showTransition: false };

  public handleChange = (name: string, value: string) => this.setState({ [name]: value });

  public handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    this.setState({ loading: true });

    const { organization, email } = this.state;

    const successCb = (done: () => void) => {
      this.setState({
        showTransition: true,
        loading: false
      });
      setTimeout(done, 1000);
    };

    await this.props.onSignup(organization, email, successCb);
  };

  public render() {
    const { organization, email, loading, showTransition } = this.state;
    const marketingText = (
      <Fragment>
        <Heading2 color={colors.tints.white}>Try out rokka for 90 days</Heading2>
        <ParagraphWhite>
          rokka is free to use for 90 days.
          <br />
          No monthly costs, no credit card required.
        </ParagraphWhite>
      </Fragment>
    );

    return (
      <SignForm isLogin={false} marketingText={marketingText} showTransition={showTransition}>
        <form onSubmit={this.handleSubmit}>
          <FormGroup label="Organization">
            <TextInput type="text" name="organization" value={organization} onChange={this.handleChange} />
          </FormGroup>
          <FormGroup label="E-mail">
            <TextInput type="email" name="email" value={email} onChange={this.handleChange} />
          </FormGroup>
          <LoadingIndicatingButton loading={loading} type="submit">
            Start free trial
          </LoadingIndicatingButton>
          <StyledTermsParagraph>
            By creating an account, you agree to rokka's
            <StyledTermsLink href="https://rokka.io/assets/pdf/Rokka_Terms_of_use_EN.pdf">
              Terms & Conditions
            </StyledTermsLink>
          </StyledTermsParagraph>
        </form>
      </SignForm>
    );
  }
}

const StyledTermsParagraph = styled.p`
  font-size: ${fonts.Sizes.small};
  color: ${colors.gray.darkest};
`;

const StyledTermsLink = styled(Link)`
  font-size: ${fonts.Sizes.small};
  margin-left: ${spaces.xsmall};
`;
