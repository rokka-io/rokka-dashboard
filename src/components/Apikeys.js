import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { authRequired } from '../utils/auth'
import BaseLayout from './layouts/BaseLayout'
import rokka from '../rokka'
import ApikeyRow from './ApikeyRow'

const DEFAULT_STATE = {
  loading: true,
  currentKeyId: '',
  data: [],
  showCreate: false,
  commentValue: '',
  newApiKey: null
}

class Apikeys extends PureComponent {
  constructor(props) {
    super(props)
    this.state = DEFAULT_STATE
  }

  componentDidMount() {
    this.getKeys()
  }

  getTable = data => {
    return (
      <table key={'table'}>
        <tbody>
          <tr className={'rka-h3 mb-md'}>
            <th>ID</th>
            <th>Comment</th>
            <th>Created</th>
            <th>Last Access (updated every 24h)</th>
            <th> </th>
          </tr>
          {data.map(key => {
            return (
              <ApikeyRow
                apiKey={key}
                key={key.id}
                currentKeyId={this.state.currentKeyId}
                updateKeys={this.getKeys}
              />
            )
          })}
        </tbody>
      </table>
    )
  }

  getKeys = () => {
    rokka()
      .user.getCurrentApiKey()
      .then(({ body }) => {
        this.setState({ currentKeyId: body.id })
      })
      .catch(err => {
        console.log('Could not get current key id')
      })
    rokka()
      .user.listApiKeys()
      .then(({ body }) => {
        this.setState({ loading: false, data: body })
      })
      .catch(err => {
        this.setState({ org: 'error' })
      })
  }

  showCreateNewKey = () => {
    if (this.state.showCreate) {
      rokka()
        .user.addApiKey(this.state.commentValue)
        .then(({ body }) => {
          this.setState({ showCreate: false, commentValue: '', newApiKey: body.api_key })

          this.getKeys()
        })
        .catch(err => {
          this.setState({ showCreate: false, commentValue: '' })
          alert("Api Key creation didn't work:" + err.body.error.message)
        })
    } else {
      this.setState({ showCreate: true })
    }
  }

  createNewKey = () => {}

  render() {
    return (
      <BaseLayout {...this.props}>
        <div key={'title2'} className="section rka-box no-min-height">
          <h2 className={'rka-h2 mb-md'}>Your Api Keys</h2>
        </div>
        <div key={'table'} className="section rka-box no-min-height rka-table-apikeys">
          {this.getTable(this.state.data)}
        </div>
        {this.state.newApiKey && (
          <div className="section rka-box no-min-height">
            <div className={'mb-md'}>
              Your new Api Key is:{' '}
              <span style={{ fontWeight: 'bold' }}>{this.state.newApiKey}</span>
            </div>
            <div>Please keep it somewhere safe, you can't restore it.</div>
          </div>
        )}
        <div className="section rka-box no-min-height">
          {this.state.showCreate && (
            <input
              type="text"
              placeholder={'Api Key Comment (optional)'}
              name="comment"
              value={this.state.commentValue}
              className="rka-input-txt mb-sm"
              onChange={e => this.setState({ commentValue: e.currentTarget.value })}
            />
          )}

          {this.state.showCreate && (
            <button
              className="rka-button rka-button-secondary mr-md"
              onClick={e => this.showCreateNewKey(e)}
            >
              Cancel
            </button>
          )}

          {this.state.data.length < 5 ? (
            <button className="rka-button rka-button-brand" onClick={e => this.showCreateNewKey(e)}>
              Create new Api Key
            </button>
          ) : (
            <div>
              You reached the maximum amount of Api Keys. If you want to create a new one, please
              delete some or create a new user with the same permissions.
            </div>
          )}
        </div>
      </BaseLayout>
    )
  }
}

Apikeys.propTypes = {
  router: PropTypes.shape({
    history: PropTypes.shape({
      push: PropTypes.func.isRequired
    }).isRequired
  }).isRequired,
  auth: PropTypes.object
}

export default authRequired(Apikeys)
