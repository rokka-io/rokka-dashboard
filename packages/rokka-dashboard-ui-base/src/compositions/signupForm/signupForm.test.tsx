import { shallow } from 'enzyme';
import React from 'react';
import { LoadingIndicatingButton } from '../../components';
import { TextInput } from '../../elements';
import { SignupForm } from './signupForm';

describe('Compositions', () => {
  describe('Signup', () => {
    it('renders the signup form correctly.', () => {
      const onSignup = jest.fn();
      const props = { onSignup };

      const component = shallow(<SignupForm {...props} />);

      expect(component).toMatchSnapshot();
    });
    it('calls the onSignup callback with the entered data.', () => {
      const onSignup = jest.fn();
      const props = { onSignup };

      const component = shallow(<SignupForm {...props} />);

      const inputs = component.find(TextInput);
      inputs.find('[type="text"]').simulate('change', {
        target: {
          name: 'organization',
          type: 'text',
          value: 'an-organization'
        }
      });
      inputs.find('[type="text"]').simulate('change', {
        target: {
          name: 'email',
          type: 'text',
          value: 'an-email'
        }
      });

      expect(component).toMatchSnapshot();

      component.find(LoadingIndicatingButton).simulate('click');
      expect(onSignup).toHaveBeenCalledWith('an-organization', 'an-email');
    });
  });
});
