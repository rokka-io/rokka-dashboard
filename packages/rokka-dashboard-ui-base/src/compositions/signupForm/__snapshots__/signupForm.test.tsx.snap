// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compositions Signup calls the onSignup callback with the entered data. 1`] = `
<SignForm
  isLogin={false}
  marketingText={
    <React.Fragment>
      <ForwardRef
        color="#fff"
      >
        Try out rokka for 90 days
      </ForwardRef>
      <ForwardRef>
        rokka is free to use for 90 days.
        <br />
        No monthly costs, no credit card required.
      </ForwardRef>
    </React.Fragment>
  }
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      label="Organization"
    >
      <TextInput
        name="organization"
        onChange={[Function]}
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      label="E-mail"
    >
      <TextInput
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </FormGroup>
    <LoadingIndicatingButton
      type="submit"
    >
      Start free trial
    </LoadingIndicatingButton>
    <styled.p>
      By creating an account, you agree to rokka's
      <Styled(styled.a)
        href="https://rokka.io/assets/pdf/Rokka_Terms_of_use_EN.pdf"
      >
        Terms & Conditions
      </Styled(styled.a)>
    </styled.p>
  </form>
</SignForm>
`;

exports[`Compositions Signup renders the signup form correctly. 1`] = `
<SignForm
  isLogin={false}
  marketingText={
    <React.Fragment>
      <ForwardRef
        color="#fff"
      >
        Try out rokka for 90 days
      </ForwardRef>
      <ForwardRef>
        rokka is free to use for 90 days.
        <br />
        No monthly costs, no credit card required.
      </ForwardRef>
    </React.Fragment>
  }
>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      label="Organization"
    >
      <TextInput
        name="organization"
        onChange={[Function]}
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      label="E-mail"
    >
      <TextInput
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </FormGroup>
    <LoadingIndicatingButton
      type="submit"
    >
      Start free trial
    </LoadingIndicatingButton>
    <styled.p>
      By creating an account, you agree to rokka's
      <Styled(styled.a)
        href="https://rokka.io/assets/pdf/Rokka_Terms_of_use_EN.pdf"
      >
        Terms & Conditions
      </Styled(styled.a)>
    </styled.p>
  </form>
</SignForm>
`;
